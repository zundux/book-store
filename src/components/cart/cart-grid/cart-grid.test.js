import React from "react";
import CartGrid, { EmptyCartLabel, ProductList } from "./cart-grid";

describe("Cart grid", () => {

  test("Cart grid has no products", () => {
    const products = [];
    const onRemoveFromCart = () => {
    };
    const onChangeAmountOfProductInCart = () => {
    };

    const component = shallow(
      <CartGrid {...{
        products,
        onRemoveFromCart,
        onChangeAmountOfProductInCart
      }}/>
    );
    matchSnapshot(component);
  });

  test("Cart grid has one products", () => {
    const products = [
      {
        id: 1,
        title: "product 1"
      }
    ];
    const onRemoveFromCart = () => {
    };
    const onChangeAmountOfProductInCart = () => {
    };

    const component = shallow(
      <CartGrid {...{
        products,
        onRemoveFromCart,
        onChangeAmountOfProductInCart
      }}/>
    );
    matchSnapshot(component);
  });

  test("Cart grid has a couple of products", () => {
    const products = [
      {
        id: 1,
        title: "product 1"
      }, {
        id: 2,
        title: "product 2"
      }
    ];
    const onRemoveFromCart = () => {
    };
    const onChangeAmountOfProductInCart = () => {
    };

    const component = shallow(
      <CartGrid {...{
        products,
        onRemoveFromCart,
        onChangeAmountOfProductInCart
      }}/>
    );
    matchSnapshot(component);
  });


  test("Empty cart label", () => {
    const component = shallow(
      <EmptyCartLabel/>
    );
    matchSnapshot(component);
  });


  test("Product list has a couple of products", () => {
    const products = [
      {
        id: 1,
        title: "product 1"
      }, {
        id: 2,
        title: "product 2"
      }
    ];
    const onRemoveFromCart = () => {
    };
    const onChangeAmountOfProductInCart = () => {
    };

    const component = shallow(
      <ProductList {...{
        products,
        onRemoveFromCart,
        onChangeAmountOfProductInCart
      }}/>
    );
    matchSnapshot(component);
  });
});