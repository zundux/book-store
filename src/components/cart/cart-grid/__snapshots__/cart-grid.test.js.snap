// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart grid Cart grid has a couple of products 1`] = `
<Component
  onChangeAmountOfProductInCart={[Function]}
  onRemoveFromCart={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "title": "product 1",
      },
      Object {
        "id": 2,
        "title": "product 2",
      },
    ]
  }
/>
`;

exports[`Cart grid Cart grid has no products 1`] = `<Component />`;

exports[`Cart grid Cart grid has one products 1`] = `
<Component
  onChangeAmountOfProductInCart={[Function]}
  onRemoveFromCart={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "title": "product 1",
      },
    ]
  }
/>
`;

exports[`Cart grid Empty cart label 1`] = `
<div>
  <p>
    Cart is empty
  </p>
  <p>
    Add products from 
    <Link
      replace={false}
      to="/"
    >
      catalog
    </Link>
  </p>
</div>
`;

exports[`Cart grid Product list has a couple of products 1`] = `
Array [
  <CartGridItem
    key="0"
    onChangeAmountOfProductInCart={[Function]}
    onRemoveFromCart={[Function]}
    product={
        Object {
            "id": 1,
            "title": "product 1",
          }
    }
/>,
  <CartGridItem
    key="1"
    onChangeAmountOfProductInCart={[Function]}
    onRemoveFromCart={[Function]}
    product={
        Object {
            "id": 2,
            "title": "product 2",
          }
    }
/>,
]
`;
